codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -377,41 +189,19 @@ public boolean containsTimeMillis(T name, long value) {
     }
 
     @Override
-    public boolean contains(T name, T value, Comparator<? super T> comparator) {
-        return contains(name, value, comparator, comparator);
-    }
-
-    @Override
-    public boolean contains(T name, T value,
-            Comparator<? super T> keyComparator, Comparator<? super T> valueComparator) {
-        checkNotNull(name, "name");
-        checkNotNull(value, "value");
-        checkNotNull(keyComparator, "keyComparator");
-        checkNotNull(valueComparator, "valueComparator");
-        int h = hashCodeGenerator.generateHashCode(name);
-        int i = index(h);
-        HeaderEntry e = entries.get(i);
-        while (e != null) {
-            if (e.hash == h &&
-                    keyComparator.compare(e.name, name) == 0 &&
-                    valueComparator.compare(e.value, value) == 0) {
-                return true;
+    @SuppressWarnings("unchecked")
+    public boolean contains(T name, T value, Comparator<? super T> valueComparator) {
+        Object values = map.get(name);
+        if (isList(values)) {
+            List<T> values0 = (List<T>) values;
+            for (int i = 0; i < values0.size(); i++) {
```

Comment:
I don't think that is necessary.  For `isList(..)` to be true the object must be of type `ValuesList` which extends `ArrayList`...which has a constant time `size()` method...which is an accessor...which should be inlined.